src/components/Popover/Popover.tsx

```diff
@@ -8,36 +8,81 @@ export type PopoverProps = React.HTMLAttributes<HTMLDivElement> & {
   open?: boolean;
   anchor?: 'topLeft' | 'topCenter' | 'topRight' | 'bottomLeft' | 'bottomCenter' | 'bottomRight';
   zIndex?: number;
+  autoPreview?: boolean;
 };
 
 export const POPOVER_MOUSEOUT_DELAY_MS = 200;
 
+enum Mode {
+  Open = 'open',
+  Closing = 'closing',
+  AutoOpening = 'autoOpening',
+  AutoOpen = 'autoOpen',
+  Closed = 'closed'
+}
+
 export const Popover: React.FC<PopoverProps> = ({
   className,
   children,
   content,
   open = false,
   anchor = 'topLeft',
   zIndex,
+  autoPreview = false,
   ...rest
 }) => {
-  const [isOpen, setOpen] = useState(open);
+  const [mode, setMode] = useState(() => {
+    if (open) {
+      return Mode.Open;
+    }
+
+    if (autoPreview) {
+      return Mode.AutoOpening;
+    }
+
+    return Mode.Closed;
+  });
+
   const timeout = useRef<ReturnType<typeof setTimeout> | null>(null);
 
   const handleMouseEnter = useCallback(() => {
     if (!open) {
       timeout.current && clearTimeout(timeout.current);
-      setOpen(true);
+      setMode(Mode.Open);
     }
   }, [open]);
 
   const handleMouseLeave = useCallback(() => {
     if (!open) {
-      timeout.current = setTimeout(() => setOpen(false), POPOVER_MOUSEOUT_DELAY_MS);
+      timeout.current = setTimeout(() => setMode(Mode.Closing), POPOVER_MOUSEOUT_DELAY_MS);
     }
   }, [open]);
 
-  useEffect(() => setOpen(open), [open]);
+  useEffect(() => {
+    if (mode !== Mode.Closing) return;
+
+    const closingTimeout = setTimeout(() => setMode(Mode.Closed), 750);
+
+    return () => clearTimeout(closingTimeout);
+  }, [mode]);
+
+  useEffect(() => {
+    if (mode !== Mode.AutoOpening) return;
+
+    const openingTimeout = setTimeout(() => setMode(Mode.AutoOpen), 1000);
+
+    return () => clearTimeout(openingTimeout);
+  });
```

Comment:
Should we also put a dependency array here? i.e. `[mode]`
I think it's not mandatory because this hook has to be run only once on the first render, but makes it more consistent with the other ones.